server-nodejs/routes/perms-tools.js

```diff
@@ -0,0 +1,72 @@
+/*
+ * Licensed under the GNU GPL v3
+ */
+module.exports = function(app) {
+    var db = app.locals.db;
+    var module = {};
+        module.extractIds = function(nodes) {
+            var ids = [];
+            nodes = unfoldIds(nodes);
+            for(var i in nodes) {
+                ids.push(nodes[i]._id);
+            }
+            return ids;
+        };
+
+        module.filterByAuthor = function(ids, username, callback) {
+            var perms = [];
+            db.read(ids, function(err, nodes) {
+                for(var i in nodes) {
+                    if(nodes[i] === null) {
+                        perms.push(true);
+                        continue;
+                    }
+                    if(nodes[i].author !== username) {
+                        perms.push(false);
+                        continue;
+                    }
+                    perms.push(true);
+                }
+                callback(perms);
+            });
+        };
+
+        module.filterRequest = function(table, perms) {
+            for(var i in table) {
+                if(perms[i] === false) {
+                    table.splice(i, 1, 'null');
+                }
+            }
+            return table;
+        };
+
+        module.isFromClass = function(req) {
```

Comment:
This method could have 2 parameters: `operationName` and `userClass` (these parameters are currently passed inside req). This way, it could be simple and express-agnostic. Then, what about renaming it to `isOperationAllowed` instead?